docs/adrs/0549-composite-run-steps.md

```diff
@@ -0,0 +1,273 @@
+TODO: Change file name to represent the correct PR number (PR is not created yet for this ADR)
+
+# ADR 0549: Composite Run Steps
+
+**Date**: 2020-06-17
+
+**Status**: Proposed
+
+**Relevant PR**: https://github.com/actions/runner/pull/549
+
+## Context
+Customers want to be able to compose actions from actions (ex: https://github.com/actions/runner/issues/438)
+
+An important step towards meeting this goal is to build in functionality for actions where users can simply execute any number of steps. 
+
+## Decision
+In this ADR, we support running multiple steps in an Action. In doing so, we build in support for mapping and flowing the inputs, outputs, and env variables (ex: All nested steps should have access to its parents' input variables and nested steps can overwrite the input variables).
+
+### Steps
+Example `user/composite/action.yml`
+```
+...
+using: 'composite' 
+steps:
+  - run: pip install -r requirements.txt
```

Comment:
@thboop I like this idea!

I think for the scope of this ADR (only supporting multiple run steps), I can't think of a way to refactor the starter workflows since we aren't supporting `uses` in this ADR. 

I think for the next ADR where we support `uses`, that would be a good example. 

Any ideas on a better example for multiple steps in a composite action?

Reply:
We can keep the original example then!

Nearly all of our starter workflows run `actions/checkout`, we could take one like this: https://github.com/actions/starter-workflows/blob/master/ci/c-cpp.yml and just have it run all the steps after checkout in the composite action, but I don't think we need to at the moment.